File: Backend/backend-root/itinerary-service/src/main/java/com/unibg/magellanus/backend/itinerary/service/POI.java
package com.unibg.magellanus.backend.itinerary.service;

import java.util.Objects;

public class POI {
	private double lat;
	private double lon;
	private boolean inRoute;
	
	public POI() {
	}

	public POI(double lat, double lon) {
		this.lat = lat;
		this.lon = lon;
		this.setInRoute(false);
	}

	public double getLat() {
		return lat;
	}

	public void setLat(double lat) {
		this.lat = lat;
	}

	public double getLon() {
		return lon;
	}

	public void setLon(double lon) {
		this.lon = lon;
	}

	public boolean isInRoute() {
		return inRoute;
	}

	public void setInRoute(boolean inRoute) {
		this.inRoute = inRoute;
	}

	@Override
	public int hashCode() {
		return Objects.hash(lat, lon);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		POI other = (POI) obj;
		return Double.doubleToLongBits(lat) == Double.doubleToLongBits(other.lat)
				&& Double.doubleToLongBits(lon) == Double.doubleToLongBits(other.lon);
	}

}

File: App/app/src/main/java/com/unibg/magellanus/app/itinerary/model/POI.kt
package com.unibg.magellanus.app.itinerary.model

import com.google.gson.annotations.SerializedName
import java.util.*

data class POI(
    var name: String? = null,
    val lat: Double,
    val lon: Double,
    @Transient var address: Address? = null
) {
    override fun hashCode() = Objects.hash(lat, lon)

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as POI

        if (lat != other.lat) return false
        if (lon != other.lon) return false

        return true
    }
}

data class Address(
    val country: String,
    @SerializedName(value = "countrycode")
    val countryCode: String,
    val city: String?,
    val postcode: String?,
    val county: String?,
    val street: String?,
    val state: String?
) {
    val formatted
        get() = "$street, $county, $city $postcode, $state, $country $countryCode"
}